bytecode/src/function/instructions/add_wrapped.rs

```diff
@@ -0,0 +1,178 @@
+    pub fn destination(&self) -> &Register<P> {
+        self.operation.destination()
+    }
+}
+
+impl<P: Program> Opcode for AddWrapped<P> {
+    /// Returns the opcode as a string.
+    #[inline]
+    fn opcode() -> &'static str {
+        "add.w"
+    }
+}
+
+impl<P: Program> Operation<P> for AddWrapped<P> {
+    /// Evaluates the operation.
+    #[inline]
+    fn evaluate(&self, registers: &Registers<P>) {
+        // Load the values for the first and second operands.
+        let first = match registers.load(self.operation.first()) {
+            Value::Literal(literal) => literal,
+            Value::Composite(name, ..) => P::halt(format!("{name} is not a literal")),
+        };
+        let second = match registers.load(self.operation.second()) {
+            Value::Literal(literal) => literal,
+            Value::Composite(name, ..) => P::halt(format!("{name} is not a literal")),
+        };
+
+        // Perform the operation.
+        let result = match (first, second) {
+            (Literal::I8(a), Literal::I8(b)) => Literal::I8(a.add_wrapped(&b)),
+            (Literal::I16(a), Literal::I16(b)) => Literal::I16(a.add_wrapped(&b)),
+            (Literal::I32(a), Literal::I32(b)) => Literal::I32(a.add_wrapped(&b)),
+            (Literal::I64(a), Literal::I64(b)) => Literal::I64(a.add_wrapped(&b)),
+            (Literal::I128(a), Literal::I128(b)) => Literal::I128(a.add_wrapped(&b)),
+            (Literal::U8(a), Literal::U8(b)) => Literal::U8(a.add_wrapped(&b)),
+            (Literal::U16(a), Literal::U16(b)) => Literal::U16(a.add_wrapped(&b)),
+            (Literal::U32(a), Literal::U32(b)) => Literal::U32(a.add_wrapped(&b)),
+            (Literal::U64(a), Literal::U64(b)) => Literal::U64(a.add_wrapped(&b)),
+            (Literal::U128(a), Literal::U128(b)) => Literal::U128(a.add_wrapped(&b)),
+            _ => P::halt(format!("Invalid '{}' instruction", Self::opcode())),
+        };
+
+        registers.assign(self.operation.destination(), result);
+    }
+}
+
+impl<P: Program> Parser for AddWrapped<P> {
+    type Environment = P::Environment;
+
+    /// Parses a string into an 'AddWrapped' operation.
+    #[inline]
+    fn parse(string: &str) -> ParserResult<Self> {
+        // Parse the operation from the string.
+        map(BinaryOperation::parse, |operation| Self { operation })(string)
+    }
+}
+
+impl<P: Program> fmt::Display for AddWrapped<P> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.operation)
+    }
+}
+
+impl<P: Program> FromBytes for AddWrapped<P> {
+    fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
+        Ok(Self { operation: BinaryOperation::read_le(&mut reader)? })
+    }
+}
+
+impl<P: Program> ToBytes for AddWrapped<P> {
+    fn write_le<W: Write>(&self, mut writer: W) -> IoResult<()> {
+        self.operation.write_le(&mut writer)
+    }
+}
+
+#[allow(clippy::from_over_into)]
+impl<P: Program> Into<Instruction<P>> for AddWrapped<P> {
```

Comment:
(applies to all `Into` implementations) the docs for `std` always recommend implementing `From` instead of `Into`: https://doc.rust-lang.org/std/convert/trait.Into.html (or https://doc.rust-lang.org/std/convert/trait.From.html)

Reply:
i just noticed `#[allow(clippy::from_over_into)]`, meaning this had been reported by `clippy`; what's the reason behind overriding it?

Reply:
The first few instructions were implemented this way so I followed the same pattern for the rest. @howardwu might have more insight as to why. Happy to update this if we decide to use From over Into.

Reply:
AddWrapped should not be invoked outside of this module